ModernBert Reranker not supported.

### System Info

`Alibaba-NLP/gte-reranker-modernbert-base` is not supported.

The model has been only added for embedding / headless inference. The loading for the classifier had is missing.

```
Starting FlashModernBert model on Cuda(CudaDevice(DeviceId(1)))
2025-04-18T02:45:50.805518Z ERROR text_embeddings_backend: backends/src/lib.rs:388: Could not start Candle backend: Could not start backend: cannot find tensor dense.weight
Error: Could not create backend

Caused by:
    Could not start backend: Could not start a suitable backend
make: *** [Makefile:78: start_tei] Error 1
engine-builder-py3.10root@michaelfeil-dev-pod-h100-0:~/baseten/bei# MODEL_ID=Alibaba-NLP/gte-reranker-modernbert-base make start_tei
```

### Information

- [x] Docker
- [ ] The CLI directly

### Tasks

- [x] An officially supported command
- [ ] My own modifications

### Reproduction

Using regular cli (my case on hopper) super easy to repro. 

tei-hopper-1.7

### Expected behavior

-

--- a/README.md
+++ b/README.md
@@ -101,6 +101,7 @@ Below are some examples of the currently supported models:
 | Re-Ranking         | XLM-RoBERTa | [BAAI/bge-reranker-large](https://huggingface.co/BAAI/bge-reranker-large)                                       |
 | Re-Ranking         | XLM-RoBERTa | [BAAI/bge-reranker-base](https://huggingface.co/BAAI/bge-reranker-base)                                         |
 | Re-Ranking         | GTE         | [Alibaba-NLP/gte-multilingual-reranker-base](https://huggingface.co/Alibaba-NLP/gte-multilingual-reranker-base) |
+| Re-Ranking         | ModernBert  | [Alibaba-NLP/gte-reranker-modernbert-base](https://huggingface.co/Alibaba-NLP/gte-reranker-modernbert-base) |
 | Sentiment Analysis | RoBERTa     | [SamLowe/roberta-base-go_emotions](https://huggingface.co/SamLowe/roberta-base-go_emotions)                     |
 
 ### Docker

--- a/backends/candle/src/models/modernbert.rs
+++ b/backends/candle/src/models/modernbert.rs
@@ -412,24 +412,27 @@ pub struct ModernBertClassificationHead {
 impl ModernBertClassificationHead {
     pub(crate) fn load(vb: VarBuilder, config: &ModernBertConfig) -> Result<Self> {
         let dense_weight = vb
-            .pp("dense")
+            .pp("head.dense")
             .get((config.hidden_size, config.hidden_size), "weight")?;
-        let dense_bias = vb.pp("dense").get(config.hidden_size, "bias").ok();
+        let dense_bias = vb.pp("head.dense").get(config.hidden_size, "bias").ok();
         let dense = Linear::new(
             dense_weight,
             dense_bias,
             Some(config.classifier_activation.clone()),
         );
 
-        let norm =
-            LayerNormNoBias::load(vb.pp("norm"), config.hidden_size, config.norm_eps as f32)?;
+        let norm = LayerNormNoBias::load(
+            vb.pp("head.norm"),
+            config.hidden_size,
+            config.norm_eps as f32,
+        )?;
 
-        let classifier_weight = vb.pp("dense").get(
+        let classifier_weight = vb.pp("classifier").get(
             (config.num_labels.unwrap_or(1), config.hidden_size),
             "weight",
         )?;
         let classifier_bias = vb
-            .pp("dense")
+            .pp("classifier")
             .get(config.num_labels.unwrap_or(1), "bias")
             .ok();
         let classifier = Linear::new(classifier_weight, classifier_bias, None);

--- a/backends/candle/src/models/modernbert.rs
+++ b/backends/candle/src/models/modernbert.rs
@@ -414,10 +414,9 @@ impl ModernBertClassificationHead {
         let dense_weight = vb
             .pp("head.dense")
             .get((config.hidden_size, config.hidden_size), "weight")?;
-        let dense_bias = vb.pp("head.dense").get(config.hidden_size, "bias").ok();
         let dense = Linear::new(
             dense_weight,
-            dense_bias,
+            None,
             Some(config.classifier_activation.clone()),
         );
 
@@ -433,9 +432,8 @@ impl ModernBertClassificationHead {
         )?;
         let classifier_bias = vb
             .pp("classifier")
-            .get(config.num_labels.unwrap_or(1), "bias")
-            .ok();
-        let classifier = Linear::new(classifier_weight, classifier_bias, None);
+            .get(config.num_labels.unwrap_or(1), "bias")?;
+        let classifier = Linear::new(classifier_weight, Some(classifier_bias), None);
 
         Ok(Self {
             dense,

--- a/backends/candle/tests/test_modernbert.rs
+++ b/backends/candle/tests/test_modernbert.rs
@@ -2,7 +2,9 @@ mod common;
 
 use crate::common::{sort_embeddings, SnapshotEmbeddings};
 use anyhow::Result;
-use common::{batch, cosine_matcher, download_artifacts, load_tokenizer};
+use common::{
+    batch, cosine_matcher, download_artifacts, load_tokenizer, relative_matcher, SnapshotScores,
+};
 use text_embeddings_backend_candle::CandleBackend;
 use text_embeddings_backend_core::{Backend, ModelType, Pool};
 
@@ -132,6 +134,45 @@ fn test_mini_pooled_raw() -> Result<()> {
     assert_eq!(raw_embeddings_batch[0], embeddings_single[0]);
     assert_eq!(raw_embeddings_batch[13], embeddings_single[0]);
     assert_eq!(embeddings_single.len(), 7);
+
+    Ok(())
+}
+
+#[test]
+#[serial_test::serial]
+fn test_modernbert_classification() -> Result<()> {
+    let model_root = download_artifacts("Alibaba-NLP/gte-reranker-modernbert-base", None).unwrap();
+    let tokenizer = load_tokenizer(&model_root)?;
+
+    let backend = CandleBackend::new(&model_root, "float32".to_string(), ModelType::Classifier)?;
+
+    let input_single = batch(
+        vec![tokenizer
+            .encode(
+                (
+                    "PrimeTime is a timing signoff tool",
+                    "PrimeTime can perform most accurate timing analysis",
+                ),
+                true,
+            )
+            .unwrap()],
+        [0].to_vec(),
+        vec![],
+    );
+
+    let predictions: Vec<Vec<f32>> = backend
+        .predict(input_single)?
+        .into_iter()
+        .map(|(_, v)| v)
+        .collect();
+    let predictions_single = SnapshotScores::from(predictions);
+
+    let matcher = relative_matcher();
+    insta::assert_yaml_snapshot!(
+        "modernbert_classification_single",
+        predictions_single,
+        &matcher
+    );
 
     Ok(())
 }

--- a/backends/candle/tests/test_modernbert.rs
+++ b/backends/candle/tests/test_modernbert.rs
@@ -148,13 +148,7 @@ fn test_modernbert_classification() -> Result<()> {
 
     let input_single = batch(
         vec![tokenizer
-            .encode(
-                (
-                    "PrimeTime is a timing signoff tool",
-                    "PrimeTime can perform most accurate timing analysis",
-                ),
-                true,
-            )
+            .encode(("What is Deep Learning?", "Deep Learning is not..."), true)
             .unwrap()],
         [0].to_vec(),
         vec![],

